src/bcf/report/create_report.rs

```diff
@@ -0,0 +1,295 @@
+                    attr.push_str(std::str::from_utf8(f).unwrap());
+                    s.push(attr);
+                }
+                ann_strings.push(s);
+            }
+        }
+
+        let alleles = variant.alleles();
+
+        if alleles.len() > 0 {
+            let ref_vec = alleles[0].clone();
+            let mut rfrce = String::from("");
+
+            let mut len: u8 = 0;
+
+            for c in ref_vec {
+                rfrce.push(*c as char);
+                len += 1;
+            }
+
+            for i in 1..alleles.len() {
+                let alt = alleles[i];
+                let var_string = String::from("Variant");
+                let var_type: VariantType;
+                let alternatives: Option<String>;
+                let end_position: f64;
+                let plot_start_position;
+
+                match alt {
+                    b"<DEL>" => {
+                        var_type = VariantType::Deletion;
+                        alternatives = None;
+                        end_position = end_pos.unwrap();
+                        plot_start_position = pos as f64 - 0.5;
+                    }
+                    b"<INV>" => {
+                        var_type = VariantType::Inversion;
+                        let rev: String = rfrce.chars().rev().collect();
+                        alternatives = Some(rev.clone());
+                        end_position = end_pos.unwrap();
+                        plot_start_position = pos as f64 - 0.5;
+                    }
+                    b"<DUP>" => {
+                        var_type = VariantType::Duplicate;
+                        let dup: String = [rfrce.clone(), rfrce.clone()].concat();
+                        alternatives = Some(dup.clone());
+                        end_position = end_pos.unwrap();
+                        plot_start_position = pos as f64 - 0.5;
+                    }
+                    _ => {
+                        let mut allel = String::from("");
+
+                        for c in alt {
+                            if *c as char != '<' && *c as char != '>' {
+                                allel.push(*c as char);
+                            }
+                        }
+
+                        match allel.len() {
+                            a if a < rfrce.len() => {
+                                plot_start_position = pos as f64 + 0.5; // start position + 1 due to alignment with deletions from bam (example: ref: ACTT alt: A  -> deletion is just CTT)
+                                end_position = pos as f64 - 0.5 + len as f64;
+                                var_type = VariantType::Deletion;
+                                alternatives = Some(allel.clone());
+                            }
+                            a if a > rfrce.len() => {
+                                plot_start_position = pos as f64;
+                                end_position = pos as f64 + len as f64;
```

Comment:
Why f64?

Reply:
I am using f64 in so the bases in the vega plot are displayed in the right way. If a single nucleobase is located at position 120, the corresponding vega mark is a rule that starts at 119.5 and ends at 120.5. I started out using f32 but had to change to f64 because some calculations with positions greater 84e5 were wrong messing up the plot.